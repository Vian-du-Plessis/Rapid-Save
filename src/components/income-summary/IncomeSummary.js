import React, {useState, useEffect} from 'react';

/* Import CSS */
import styles from './IncomeSummary.module.scss';

const IncomeSummary = ( props ) => {

    const [value, setValue] = useState([{name: 'Please add income'}]);

    useEffect(() => {
        setValue(props.value);
    }, [props.value])

    return (
        <div className={ styles.container }>
            <h3>{ props.title }</h3>
            <hr />
            
            <h4>{ props.subheadingOne }</h4>
            <h5>{ props.subbodyOne}</h5>

            <h4>{ props.subheadingTwo }</h4>
            <h5 className={ styles.lastElement }>{ props.subbodyTwo }</h5>
        </div>
    );
};

export default IncomeSummary;