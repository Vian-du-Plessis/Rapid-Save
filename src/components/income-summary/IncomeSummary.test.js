import IncomeSummary from "./IncomeSummary";
import { render, screen } from "@testing-library/react";
import userEvent from '@testing-library/user-event';
import "@testing-library/jest-dom/extend-expect";
import renderer from 'react-test-renderer';


const IncomeSummaryProps = {
    title: 'Household income summary',
    subheadingOne: 'Total Income',
    subbodyOne: 'R 0',
    subheadingTwo: 'Average Income',
    subbodyTwo: 'R 0'
}

describe("Test Income Summary Component...", () => {

    test("Show correct title...", () => {
        render(
            <IncomeSummary
                {...IncomeSummaryProps}
            />
        );
        const titleElement = screen.getByText("Household income summary");

        expect(titleElement).toBeInTheDocument();
        expect(titleElement.textContent).toBe("Household income summary");
    });

    test("Show correct sub Heading One...", () => {
        render(
            <IncomeSummary
                {...IncomeSummaryProps}
            />
        );
        const titleElement = screen.getByText("Total Income");

        expect(titleElement).toBeInTheDocument();
        expect(titleElement.textContent).toBe("Total Income");
    });

    test("Show correct sub Heading Two...", () => {
        render(
            <IncomeSummary
                {...IncomeSummaryProps}
            />
        );
        const titleElement = screen.getByText("Average Income");

        expect(titleElement).toBeInTheDocument();
        expect(titleElement.textContent).toBe("Average Income");
    });

    // SNAPSHOT TEST
    test("Snapshot of component", () => {
        const component = renderer.create(
            <IncomeSummary/>
        );

        let tree = component.toJSON();

        expect(tree).toMatchSnapshot();

        tree = component.toJSON();
        expect(tree).toMatchSnapshot()
    })

})