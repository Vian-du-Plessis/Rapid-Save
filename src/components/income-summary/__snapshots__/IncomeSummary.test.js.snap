// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Income Summary Component...  Snapshot of component 1`] = `
<div
  className="container"
>
  <h3 />
  <hr />
  <h4 />
  <h5 />
  <h4 />
  <h5
    className="lastElement"
  />
</div>
`;

exports[`Test Income Summary Component...  Snapshot of component 2`] = `
<div
  className="container"
>
  <h3 />
  <hr />
  <h4 />
  <h5 />
  <h4 />
  <h5
    className="lastElement"
  />
</div>
`;
